[COST-5805] Update docs with what's new in v4.0.0 (#581)
project-koku/koku-metrics-operator: docs/csv-description.md

### NOTE:
  * The API name change only impacts users of the upstream (community) Koku Metrics Operator.
  * The downstream (Red Hat-supported) Cost Management Metrics Operator is not impacted by this change and users should continue using their existing configurations.

**Important Upgrade Instructions:**

  If you are upgrading the upstream Koku Metrics Operator to version 4.0.0 or higher, you must manually migrate your configuration. The operator will no longer recognize existing KokuMetricsConfig resources.

  To successfully upgrade and retain your operator's configuration:

  1. Backup Existing Configuration (Optional but Recommended):

     Before upgrading, retrieve the details of your current KokuMetricsConfig instance. This will help you recreate it with the new API name.

     ```
     oc get kokumetricsconfig -n koku-metrics-operator <your-config-name> -o yaml > koku-metrics-config-backup.yaml
     ```
  2. Delete the Old KokuMetricsConfig Instance:

>       Once the operator has been upgraded to v4.0.0, you must delete any existing KokuMetricsConfig resources.

Author: You're right to question the strong wording here 😁. But yes, deleting the old `KokuMetricsConfig` should be recommended even though the upgraded operator won't reconcile it. 
I've updated the docs.

>       Once the operator has been upgraded to `version 4.0.0`, delete any existing KokuMetricsConfig resources.

Reviewer: ```suggestion
      After the operator is upgraded to `version 4.0.0`, delete any existing `KokuMetricsConfig` resources.
```

      ```
      oc delete kokumetricsconfig -n koku-metrics-operator <your-config-name>
      ```

  3. Create the New CostManagementMetricsConfig Instance:

>       Using the information from your backup (or a new configuration), create a new CostManagementMetricsConfig instance. Adjust the spec section based on your specific needs. [Refer to the configurable parameters](#configurable-parameters).

Author: No, the spec is part of the configuration.
